lib/Basics/VelocyPackHelper.cpp

```diff
@@ -387,6 +391,188 @@ int VelocyPackHelper::compareNumberValues(VPackValueType lhsType,
   return (l < r ? -1 : 1);
 }
 
+template<typename T>
+int comp(T a, T b) {
+  if (a == b) {
+    return 0;
+  }
+  return (a < b ? -1 : 1);
+}
```

Comment:
I have played around with it, but was not really satisfied, since the casting and asserting seems to be more trouble than the rewards we can reap. If at all, it would be the `std::weak_ordering`, since values which are considered equal are distinguishable.